packages/trader/src/AppV2/Components/CurrentSpot/current-spot.tsx

```diff
@@ -0,0 +1,157 @@
+    const is_digit_contract_ended = is_ended && is_digit_contract;
+    const is_won = is_digit_contract_ended && status === STATUS.WON;
+    const is_lost = is_digit_contract_ended && status === STATUS.LOST;
+    const digits_array = Object.keys(digits_info)
+        .sort((a, b) => +a - +b)
+        .map(spot_time => digits_info[+spot_time]);
+    // last_contract_digit refers to digit and spot values from last digit contract in contracts array:
+    const last_contract_digit = React.useMemo(() => digits_array.slice(-1)[0] || {}, [digits_array]);
+    const latest_tick_pip_size = tick ? +tick.pip_size : null;
+    const latest_tick_quote_price =
+        tick?.quote && latest_tick_pip_size ? tick.quote.toFixed(latest_tick_pip_size) : null;
+    const latest_tick_digit = latest_tick_quote_price ? +(latest_tick_quote_price.split('').pop() || '') : null;
+    // latest_digit refers to digit and spot values from the latest price:
+    const latest_digit = React.useMemo(
+        () =>
+            is_won || is_lost
+                ? (last_contract_digit as { digit: number | null; spot: string | null })
+                : { digit: latest_tick_digit, spot: latest_tick_quote_price },
+        [is_won, is_lost, latest_tick_digit, latest_tick_quote_price, last_contract_digit]
+    );
+
+    const [displayed_tick, setDisplayedTick] = React.useState<number | null>(current_tick);
+    const [displayed_spot, setDisplayedSpot] = React.useState<string | null>(latest_digit.spot);
+    const [should_enter_from_top, setShouldEnterFromTop] = React.useState(false);
+
+    const barrier = !is_contract_elapsed && !!tick ? Number(contract_info.barrier) : null;
+    const is_winning = isDigitContractWinning(contract_type, barrier, latest_digit.digit);
+    const has_contract = is_digit_contract && status && latest_digit.spot && !!entry_tick;
+    const has_open_contract = has_contract && !is_ended;
+    const has_relevant_tick_data = underlying === symbol || !underlying;
+    const should_show_tick_count = has_contract && has_relevant_tick_data;
+    const should_enter_from_left =
+        !prev_contract?.contract_info ||
+        !!(prev_contract?.contract_info?.is_sold && last_contract.contract_info?.tick_stream?.length === 1);
+
+    const setNewData = React.useCallback(() => {
+        setDisplayedTick(current_tick);
+        setDisplayedSpot(latest_digit.spot);
+    }, [current_tick, latest_digit.spot]);
+
+    React.useEffect(() => {
+        const has_multiple_contracts =
+            prev_contract?.contract_info &&
+            !prev_contract?.contract_info?.is_sold &&
+            last_contract.contract_info?.entry_tick;
+        const is_next_contract_opened = prev_contract_id && contract_id && prev_contract_id !== contract_id;
+        if (has_multiple_contracts && is_next_contract_opened) {
+            setShouldEnterFromTop(true);
+            contract_switching_timer.current = setTimeout(() => {
```

Comment:
Could we clean up this timeout every time on this useEffect's cleanup instead of cleaning up just once on the unmount? I think there will be a problem in the animation when multiple contracts are purchased within the animation duration.

Reply:
@balakrishna-deriv no, unfortunately, animation will break in this case because setTimeout will get cleared on every useEffect code execution (there are too many dep's and it updates quite often) while we still need to execute the code inside the setTimeout on each useEffect code execution because it's very important to update data and turn off the animation after every update. We need to clear the setTimeout only onUnmount. And there are no issues with the current multiple contracts animation, please check the test link to verify. :)